packages/ember-glimmer/tests/utils/shared-conditional-tests.js

```diff
@@ -349,6 +351,59 @@ export const ObjectTestCases = {
     });
 
     this.assertText('T1T2F3');
+  },
+
+  ['@test setting the isTruthy property of an object proxy overrides its truthiness regardless of its content']() {
```

Comment:
Is this a test we actually want? /cc @wycats

This test seems terrible. At best this is intimate API (private API that has become popular to abuse) and should be clearly marked as such.